### PR TITLE
feat(router): output that would notify when link is activated or inactivated

### PR DESCRIPTION
**Pull request by anandtiwary**
feat(router): output that would notify when link is activated or inactivated

This commit adds a new output to `routerLinkActive` directive.
Whenever, the associated link becomes active or inactive, an
event will be fired on this out with the correct status

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: https://github.com/angular/angular/issues/37284

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by anandtiwary** [comment]
@atscott Could you please point me to any documentation for running these lint and test commands locally ?

**Comment by atscott** [comment]
@anandtiwary 
The lint check is checking the [commit message guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit).

To fix the API test, run `yarn bazel run //packages/router:router_api.accept`
The command to run this group of tests is `yarn bazel test //... --build_tag_filters=-ivy-only --test_tag_filters=-ivy-only`. 
There's not documentation on why, when, or how to run these API tests. It's not really important that you run them yourself. It's only important that they're run on CI and the failures tell you how to resolve them.

**Review by atscott — CHANGES_REQUESTED** [review]
I don't necessarily disagree with some of the styling changes in this PR, but could you please separate them out into a different PR or commit to make reviewing more straightforward?

**Comment by anandtiwary** [comment]
> @anandtiwary 
> 
> The lint check is checking the [commit message guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit).
> 
> 
> 
> To fix the API test, run `yarn bazel run //packages/router:router_api.accept`
> 
> The command to run this group of tests is `yarn bazel test //... --build_tag_filters=-ivy-only --test_tag_filters=-ivy-only`. 
> 
> There's not documentation on why, when, or how to run these API tests. It's not really important that you run them yourself. It's only important that they're run on CI and the failures tell you how to resolve them.

Thanks! Can we add a hook to run this lint check at commit time or befofe pushing ?

**Comment by atscott** [comment]
@anandtiwary There actually already is a hook for this. It used to be a failure but that caused other problems. There was probably a warning that you didn't notice when you added the commit that didn't conform to the requirements.

File: packages/router/src/directives/router_link_active.ts [hunk 1]
```diff
@@ -163,24 +180,28 @@ export class RouterLinkActive implements OnChanges, OnDestroy, AfterContentInit
...
   }
 
-  private isLinkActive(router: Router): (link: (RouterLink|RouterLinkWithHref)) => boolean {
+  private isLinkActive(router: Router): (link: RouterLink|RouterLinkWithHref) => boolean {
     const options: boolean|IsActiveMatchOptions =
         isActiveMatchOptions(this.routerLinkActiveOptions) ?
-        this.routerLinkActiveOptions :
-        // While the types should disallow `undefined` here, it's possible without strict inputs
-        (this.routerLinkActiveOptions.exact || false);
+        this.routerLinkActiveOptions  // While the types should disallow `undefined` here, it's
```
Comment (anandtiwary) [inline packages/router/src/directives/router_link_active.ts:193]: All these are formatting which i think automatically came up after commit.

**Comment by anandtiwary** [comment]
Shouldn't the lint job ignore the merge master commit?

**Comment by anandtiwary** [comment]
How do you recommend i fix this?

**Comment by alfaproject** [comment]
You should rebase with latest master and squash your commits into just 1: https://levelup.gitconnected.com/how-to-squash-git-commits-9a095c1bc1fc

**Comment by anandtiwary** [comment]
> You should rebase with latest master and squash your commits into just 1: https://levelup.gitconnected.com/how-to-squash-git-commits-9a095c1bc1fc

Yeah, but I have the bad commit in the remote branch and in this PR. So, wouldn't force amend be a better choice?

**Comment by alfaproject** [comment]
Amend only mutates the last commit. You need to squash everything into 1 commit as per Angular contribution guide

**Comment by anandtiwary** [comment]
I think I messed it up. will create a new PR

**Comment by anandtiwary** [comment]
@atscott Sorry for messing up the PR. Please check https://github.com/angular/angular/pull/43188

**Pull request closed without merging**